diff --git a/cups/http.h b/cups/http.h
index 6c45301e..ac166970 100644
--- a/cups/http.h
+++ b/cups/http.h
@@ -556,10 +556,11 @@ extern int		httpSetCredentials(http_t *http, cups_array_t *certs) _CUPS_API_1_5;
 extern void		httpSetTimeout(http_t *http, double timeout, http_timeout_cb_t cb, void *user_data) _CUPS_API_1_5;
 
 /**** New in CUPS 1.6/macOS 10.8 ****/
-extern http_addrlist_t	*httpAddrConnect2(http_addrlist_t *addrlist, int *sock, int msec, int *cancel) _CUPS_API_1_6;
+extern http_addrlist_t	*httpAddrConnect2(http_addrlist_t *addrlist, int *sock, int msec, int *cancel, const char *nic) _CUPS_API_1_6;
 extern http_state_t	httpGetState(http_t *http) _CUPS_API_1_6;
 extern http_version_t	httpGetVersion(http_t *http) _CUPS_API_1_6;
 extern int		httpReconnect2(http_t *http, int msec, int *cancel) _CUPS_API_1_6;
+extern int		httpReconnect3(http_t *http, int msec, int *cancel, const char *nic) _CUPS_API_1_6;
 
 
 /**** New in CUPS 1.7/macOS 10.9 ****/
@@ -569,6 +570,7 @@ extern int		httpAddrListen(http_addr_t *addr, int port) _CUPS_API_1_7;
 extern int		httpAddrPort(http_addr_t *addr) _CUPS_API_1_7;
 extern char		*httpAssembleUUID(const char *server, int port, const char *name, int number, char *buffer, size_t bufsize) _CUPS_API_1_7;
 extern http_t		*httpConnect2(const char *host, int port, http_addrlist_t *addrlist, int family, http_encryption_t encryption, int blocking, int msec, int *cancel) _CUPS_API_1_7;
+extern http_t		*httpConnect3(const char *host, int port, http_addrlist_t *addrlist, int family, http_encryption_t encryption, int blocking, int msec, int *cancel, const char *nic) _CUPS_API_1_7;
 extern const char	*httpGetContentEncoding(http_t *http) _CUPS_API_1_7;
 extern http_status_t	httpGetExpect(http_t *http) _CUPS_API_1_7;
 extern ssize_t		httpPeek(http_t *http, char *buffer, size_t length) _CUPS_API_1_7;
