diff --git a/cups-2.4.0/backend/ipp.c b/cups-2.4.0/backend/ipp.c
index c4b34668..60b867a0 100644
--- a/cups-2.4.0/backend/ipp.c
+++ b/cups-2.4.0/backend/ipp.c
@@ -64,6 +64,7 @@ typedef struct _cups_monitor_s		/**** Monitoring data ****/
   ipp_jstate_t		job_state;	/* Current job state */
   ipp_pstate_t		printer_state;	/* Current printer state */
   int			retryable;	/* Is this a job that should be retried? */
+  const char *nic;
 } _cups_monitor_t;
 
 
@@ -273,6 +274,7 @@ main(int  argc,				/* I - Number of command-line args */
   ppd_file_t	*ppd = NULL;		/* PPD file */
   _ppd_cache_t	*pc = NULL;		/* PPD cache and mapping data */
   fd_set	input;			/* Input set for select() */
+  const char *nic = NULL;
 
 
  /*
@@ -651,6 +653,10 @@ main(int  argc,				/* I - Number of command-line args */
     password = getenv("AUTH_PASSWORD");
   }
 
+  num_options = cupsParseOptions(argv[5], 0, &options);
+  nic = cupsGetOption("nic", num_options, options);
+  fprintf(stderr, "DEBUG: nic %s\n", nic);
+
  /*
   * Try finding the remote server...
   */
@@ -659,8 +665,11 @@ main(int  argc,				/* I - Number of command-line args */
 
   addrlist = backendLookup(hostname, port, &job_canceled);
 
-  http = httpConnect2(hostname, port, addrlist, AF_UNSPEC, cupsEncryption(), 1,
-                      0, NULL);
+  if (nic != NULL) {
+    http = httpConnect3(hostname, port, addrlist, AF_UNSPEC, cupsEncryption(), 1, 0, NULL, nic);
+  } else {
+    http = httpConnect2(hostname, port, addrlist, AF_UNSPEC, cupsEncryption(), 1, 0, NULL);
+  }
   httpSetTimeout(http, 30.0, timeout_cb, NULL);
 
  /*
@@ -701,7 +710,7 @@ main(int  argc,				/* I - Number of command-line args */
     fprintf(stderr, "DEBUG: Connecting to %s:%d\n", hostname, port);
     _cupsLangPrintFilter(stderr, "INFO", _("Connecting to printer."));
 
-    if (httpReconnect2(http, 30000, NULL))
+    if (nic != NULL ? httpReconnect3(http, 30000, NULL, nic) : httpReconnect2(http, 30000, NULL))
     {
       int error = errno;		/* Connection error */
 
@@ -986,7 +995,7 @@ main(int  argc,				/* I - Number of command-line args */
 	  version = 10;
         }
 
-	httpReconnect2(http, 30000, NULL);
+	nic != NULL ? httpReconnect3(http, 30000, NULL, nic) : httpReconnect2(http, 30000, NULL);
       }
       else if (ipp_status == IPP_STATUS_ERROR_NOT_FOUND)
       {
@@ -1018,7 +1027,7 @@ main(int  argc,				/* I - Number of command-line args */
 	                     _("Unable to get printer status."));
         sleep(10);
 
-	httpReconnect2(http, 30000, NULL);
+	nic != NULL ? httpReconnect3(http, 30000, NULL, nic) : httpReconnect2(http, 30000, NULL);
       }
 
       ippDelete(supported);
@@ -1450,6 +1459,7 @@ main(int  argc,				/* I - Number of command-line args */
   monitor.job_state     = IPP_JSTATE_PENDING;
   monitor.printer_state = IPP_PSTATE_IDLE;
   monitor.retryable     = argc == 6 && document_format && strcmp(document_format, "image/pwg-raster") && strcmp(document_format, "image/urf");
+  monitor.nic           = nic;
 
   fprintf(stderr, "DEBUG: retryable=%d\n", monitor.retryable);
 
@@ -2063,7 +2073,7 @@ main(int  argc,				/* I - Number of command-line args */
       * Do the request...
       */
 
-      httpReconnect2(http, 30000, NULL);
+      nic != NULL ? httpReconnect3(http, 30000, NULL, nic) : httpReconnect2(http, 30000, NULL);
       response   = cupsDoRequest(http, request, resource);
       ipp_status = cupsLastError();
 
@@ -2457,8 +2467,13 @@ monitor_printer(
   * Make a copy of the printer connection...
   */
 
-  http = httpConnect2(monitor->hostname, monitor->port, NULL, AF_UNSPEC,
-                      monitor->encryption, 1, 0, NULL);
+  if (monitor->nic != NULL) {
+    http = httpConnect3(monitor->hostname, monitor->port, NULL, AF_UNSPEC,
+                        monitor->encryption, 1, 0, NULL, monitor->nic);
+  } else {
+    http = httpConnect2(monitor->hostname, monitor->port, NULL, AF_UNSPEC,
+                        monitor->encryption, 1, 0, NULL);
+  }
   httpSetTimeout(http, 30.0, timeout_cb, NULL);
   if (username[0])
     cupsSetUser(username);
@@ -2480,7 +2495,7 @@ monitor_printer(
     */
 
     if (httpGetFd(http) < 0)
-      httpReconnect2(http, 30000, NULL);
+      monitor->nic != NULL ? httpReconnect3(http, 30000, NULL, monitor->nic) : httpReconnect2(http, 30000, NULL);
 
     if (httpGetFd(http) >= 0)
     {
@@ -2702,7 +2717,7 @@ monitor_printer(
   if (job_canceled > 0 && monitor->job_id > 0)
   {
     if (httpGetFd(http) < 0)
-      httpReconnect2(http, 30000, NULL);
+      monitor->nic != NULL ? httpReconnect3(http, 30000, NULL, monitor->nic) : httpReconnect2(http, 30000, NULL);
 
     if (httpGetFd(http) >= 0)
     {
