diff --git a/cups/ppd-cache.c b/cups/ppd-cache.c
index a3235448..5dca647d 100644
--- a/cups/ppd-cache.c
+++ b/cups/ppd-cache.c
@@ -210,7 +210,7 @@ _cupsConvertOptions(
 
   media_source = _ppdCacheGetSource(pc, cupsGetOption("InputSlot", num_options, options));
   media_type   = _ppdCacheGetType(pc, cupsGetOption("MediaType", num_options, options));
-  size         = _ppdCacheGetSize(pc, keyword, /*ppd_size*/NULL);
+  size         = _ppdCacheGetSize2(pc, keyword, /*ppd_size*/NULL);
 
   if (media_col_sup && (size || media_source || media_type))
   {
@@ -2769,6 +2769,19 @@ _ppdCacheGetPageSize(
 
 pwg_size_t *				/* O - PWG size or NULL */
 _ppdCacheGetSize(
+    _ppd_cache_t *pc,			/* I - PPD cache and mapping data */
+    const char   *page_size)		/* I - PPD PageSize */
+{
+  return (_ppdCacheGetSize2(pc, page_size, NULL));
+}
+
+
+/*
+ * '_ppdCacheGetSize2()' - Get the PWG size associated with a PPD PageSize.
+ */
+
+pwg_size_t *				/* O - PWG size or NULL */
+_ppdCacheGetSize2(
     _ppd_cache_t *pc,			/* I - PPD cache and mapping data */
     const char   *page_size,		/* I - PPD PageSize */
     ppd_size_t   *ppd_size)		/* I - PPD page size information */
diff --git a/cups/ppd-private.h b/cups/ppd-private.h
index 44f26bf4..9cea8069 100644
--- a/cups/ppd-private.h
+++ b/cups/ppd-private.h
@@ -181,6 +181,8 @@ extern const char	*_ppdCacheGetOutputBin(_ppd_cache_t *pc,
 extern const char	*_ppdCacheGetPageSize(_ppd_cache_t *pc, ipp_t *job,
 			                      const char *keyword, int *exact) _CUPS_PRIVATE;
 extern pwg_size_t	*_ppdCacheGetSize(_ppd_cache_t *pc,
+			                  const char *page_size) _CUPS_PRIVATE;
+extern pwg_size_t	*_ppdCacheGetSize2(_ppd_cache_t *pc,
 			                  const char *page_size, ppd_size_t *ppd_size) _CUPS_PRIVATE;
 extern const char	*_ppdCacheGetSource(_ppd_cache_t *pc,
 			                    const char *input_slot) _CUPS_PRIVATE;
diff --git a/cups/testppd.c b/cups/testppd.c
index b9b584d1..c72a65fe 100644
--- a/cups/testppd.c
+++ b/cups/testppd.c
@@ -1369,7 +1369,7 @@ main(int  argc,				/* I - Number of command-line arguments */
       {
         pwg_size_t	*pwg;		/* PWG media size */
 
-	pwg = _ppdCacheGetSize(pc, size->name, size);
+	pwg = _ppdCacheGetSize2(pc, size->name, size);
 
         printf("    media-default: %s\n", pwg ? pwg->map.pwg : "unknown");
       }
diff --git a/filter/rastertopwg.c b/filter/rastertopwg.c
index b0edba61..f8107595 100644
--- a/filter/rastertopwg.c
+++ b/filter/rastertopwg.c
@@ -274,7 +274,7 @@ main(int  argc,				/* I - Number of command-line args */
       }
     }
 
-    if (inheader.cupsPageSizeName[0] && (pwg_size = _ppdCacheGetSize(cache, inheader.cupsPageSizeName, /*ppd_size*/NULL)) != NULL && pwg_size->map.pwg)
+    if (inheader.cupsPageSizeName[0] && (pwg_size = _ppdCacheGetSize2(cache, inheader.cupsPageSizeName, /*ppd_size*/NULL)) != NULL && pwg_size->map.pwg)
     {
       strlcpy(outheader.cupsPageSizeName, pwg_size->map.pwg,
 	      sizeof(outheader.cupsPageSizeName));
diff --git a/scheduler/printers.c b/scheduler/printers.c
index 24951068..9b176035 100644
--- a/scheduler/printers.c
+++ b/scheduler/printers.c
@@ -4194,7 +4194,7 @@ load_ppd(cupsd_printer_t *p)		/* I - Printer */
       */
 
       if ((size = ppdPageSize(ppd, NULL)) != NULL)
-	pwgsize = _ppdCacheGetSize(p->pc, size->name, size);
+	pwgsize = _ppdCacheGetSize2(p->pc, size->name, size);
       else
         pwgsize = NULL;
 
